components/kyma-operator/pkg/kymainstallation/step.go

```diff
@@ -60,7 +60,31 @@ func (s installStep) Run() error {
 		releaseOverrides)
 
 	if installErr != nil {
-		return errors.New("Helm install error: " + installErr.Error())
+		installErrMsg := fmt.Sprintf("Helm install error: %s", installErr.Error())
+		errorMsg := installErrMsg
+
+		isDeletable, err := s.helmClient.IsReleaseDeletable(s.component.GetReleaseName())
+		if err != nil {
+			errMsg := fmt.Sprintf("Checking status of %s failed with an error: %s", s.component.GetReleaseName(), err.Error())
+			log.Println(errMsg)
+			return errors.New(fmt.Sprintf("%s \n %s \n", installErrMsg, errMsg))
+		}
+
+		if isDeletable {
+			log.Println(fmt.Sprintf("Helm installation of %s failed. Deleting before retrying installation.", s.component.GetReleaseName()))
+			log.Println("Look at the proceeding logs to see the reason of failure")
+			_, err := s.helmClient.DeleteRelease(s.component.GetReleaseName())
+
+			if err != nil {
+				deleteErrMsg := fmt.Sprintf("Helm delete of %s failed with an error: %s", s.component.GetReleaseName(), err.Error())
+				log.Println(deleteErrMsg)
+				return errors.New(fmt.Sprintf("%s \n %s \n", installErrMsg, deleteErrMsg))
+			}
+
+			errorMsg = fmt.Sprintf("%s\nHelm delete of %s was successfull", installErrMsg, s.component.GetReleaseName())
```

Comment:
We may need to wait here after delete command. `helm del` reports status but it does not always mean that release is fully deleted. Release has status deleting and if we start installation quickly enough, we will have error that release is being deleted, another delete and another retry but we increase the delay between retries.

Reply:
introduced hardcoded 5s sleep, in the future we might think of more sophisticated solution

Reply:
What if instead of this fixed 5s sleep we put there a function that gets the release status (we have such code already) and returns true when helm reports "release does not exist" (we have such code already)? Then we wrap this into a retry.Do and perform it 25 times with 5 sec fixed delay. This is a simple function, about 20 loc. and it gives us way more flexibility: 5 sec resolution up to 2 minutes of waiting.